src/Perl6/Metamodel/EnumHOW.nqp

```diff
@@ -85,13 +88,18 @@ class Perl6::Metamodel::EnumHOW
     }
 
     method enum_from_value($obj, $value) {
-        unless %!value_to_enum {
+        my $value_to_enum := $!value_to_enum;
+        unless $value_to_enum {
+            my $value_to_enum := nqp::hash;
```

Comment:
I don't think this should be redeclared `my` here, otherwise the outer `$value_to_enum` will be `NQPMu` the first time this is called (I'm surprised it doesn't crash in some way). Otherwise, it's looking fine.

Reply:
Hah, yes, apparently it's fine!

```
$ nqp -e 'my $x := nqp::hash; say($x.HOW.name($x))'
BOOTHash
```